citesphere-importer/src/test/java/edu/asu/diging/citesphere/importer/core/service/CrossrefReferenceImportProcessorTest.java

```diff
@@ -0,0 +1,112 @@
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import edu.asu.diging.citesphere.importer.core.exception.IteratorCreationException;
+import edu.asu.diging.citesphere.importer.core.kafka.impl.KafkaRequestProducer;
+import edu.asu.diging.citesphere.importer.core.model.BibEntry;
+import edu.asu.diging.citesphere.importer.core.model.ItemType;
+import edu.asu.diging.citesphere.importer.core.model.impl.Publication;
+import edu.asu.diging.citesphere.importer.core.service.impl.CrossrefReferenceImportProcessor;
+import edu.asu.diging.citesphere.importer.core.service.impl.JobInfo;
+import edu.asu.diging.citesphere.importer.core.service.parse.BibEntryIterator;
+import edu.asu.diging.citesphere.importer.core.service.parse.IHandlerRegistry;
+import edu.asu.diging.citesphere.importer.core.zotero.IZoteroConnector;
+import edu.asu.diging.citesphere.importer.core.zotero.template.IJsonGenerationService;
+import edu.asu.diging.citesphere.messages.model.KafkaJobMessage;
+import edu.asu.diging.citesphere.messages.model.ResponseCode;
+import edu.asu.diging.citesphere.messages.model.Status;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CrossrefReferenceImportProcessorTest {
+
+    @Spy
+    @InjectMocks
+    private CrossrefReferenceImportProcessor processor;
+
+    @Mock
+    private IZoteroConnector zoteroConnector;
+
+    @Mock
+    private IJsonGenerationService generationService;
+
+    @Mock
+    private IHandlerRegistry handlerRegistry;
+
+    @Mock
+    private BibEntryIterator bibIterator;
+
+    @Mock
+    private KafkaJobMessage message;
+
+    @Mock
+    private JobInfo info;
+
+    @Mock
+    private KafkaRequestProducer requestProducer;
+
+    private ObjectMapper mapper;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        processor.init();
+        mapper = new ObjectMapper();
+        when(message.getId()).thenReturn("testMessageId");
+        doNothing().when(processor).sendMessage(any(), anyString(), any(Status.class), any(ResponseCode.class));
+    }
+
+    @Test
+    public void testStartImport_successful() throws Exception {
+        when(info.getDois()).thenReturn(Arrays.asList("10.1234/example1", "10.5678/example2"));
+        when(handlerRegistry.handleFile(info, null)).thenReturn(bibIterator);
```

Comment:
the crossref importer doesn't have a handlerregistry, does it?

Reply:
Deleted the tests as CrossrefRefernceImport does not have much now.